### PR TITLE
Add expected bucket owner configuration

### PR DESCRIPTION
**Pull request by sum12**
## Description of change

#286 

## Does this change impact existing behavior?

No

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3-client/src/s3_crt_client.rs [hunk 1]
```diff
@@ -868,4 +885,13 @@ mod tests {
...
     }
+
+    /// Simple test to ensure the expected bucket owner can be set
+    #[test]
+    fn test_expected_bucket_owner() {
+        let mut config: S3ClientConfig = Default::default();
+
+        config = config.bucket_owner("some_acnt_id");
+        assert_eq!(config.bucket_owner, Some("some_acnt_id".to_owned()));
+    }
```
Comment (monthonk) [inline mountpoint-s3-client/src/s3_crt_client.rs:896]: We probably want to verify the value in the header too, like what we do in `test_user_agent`.
```suggestion
    #[test]
    fn test_expected_bucket_owner() {
        let expected_bucket_owner = "111122223333";

        let config: S3ClientConfig = S3ClientConfig::new().bucket_owner("111122223333");

        let client = S3CrtClient::new("eu-west-1", config).expect("Create test client");

        let mut message = client
            .inner
            .new_request_template("GET", "doc-example-bucket")
            .expect("new request template expected");

        let headers = message.inner.get_headers().expect("Expected a block of HTTP headers");

        let expected_bucket_owner_header = headers
            .get("x-amz-expected-bucket-owner")
            .expect("the headers should contain x-amz-expected-bucket-owner");
        let expected_bucket_owner_value = expected_bucket_owner_header.value();

        assert!(expected_bucket_owner_value
            .to_string_lossy()
            .starts_with(expected_bucket_owner));
    }
```

**Review by monthonk — CHANGES_REQUESTED** [review]
Thanks for your contribution! I think we should extend the test a bit to also verify the http headers and a few nit-picking, but overall looks good to me.

File: mountpoint-s3/src/main.rs [hunk 2]
```diff
@@ -234,6 +234,9 @@ struct CliArgs {
 
     #[clap(long, help = "Use a specific profile from your credential file.", help_heading = AWS_CREDENTIALS)]
     pub profile: Option<String>,
+
+    #[clap(long, help = "Expected bucket owner (use: <aws-acct-id>)", help_heading = AWS_CREDENTIALS)]
```
[the remark this answers is not in the document]
Comment (dannycjones) [reply mountpoint-s3/src/main.rs:238]: Even better, we can use [`value_name`](
https://docs.rs/clap/latest/clap/struct.Arg.html#method.value_name) to replace the value placeholder directly. (Plus adding some more detailed help text.):

```suggestion
    #[clap(
        long,
        help = "Account ID of the expected bucket owner. \
            If the bucket is owned by a different account, S3 requests fail with an access denied error.",
        help_heading = AWS_CREDENTIALS,
        value_name = "AWS_ACCOUNT_ID"
    )]
    pub expected_bucket_owner: Option<String>,
```

Output is like below:

```
AWS credentials options:
      --expected-bucket-owner <AWS_ACCOUNT_ID>
          Account ID of the expected bucket owner. If the bucket is owned by a different account, the S3 requests fail with an access denied error.
```

**Comment by dannycjones** [comment]
I updated the argument definition for the placeholder and help text. I also merged in latest `main` commit so CI should be passing as expected.

**Review by monthonk — APPROVED** [review]

**Pull request merged**